pkg/aws/aws_client/role.go

```diff
@@ -48,6 +48,19 @@ func (client *AWSClient) CreateRole(roleName string,
 	return *resp.Role, err
 }
 
+func (client *AWSClient) CreateRoleAndAttachPolicy(roleName string,
+	assumeRolePolicyDocument string,
+	permissionBoundry string,
+	tags map[string]string,
+	path string,
+	policyArn string) (types.Role, error) {
+	role, err := client.CreateRole(roleName, assumeRolePolicyDocument, permissionBoundry, tags, path)
+	if err != nil {
```

Comment:
@OAharoni-RedHat Do you need to check the `err` before you try and call `client.AttachPolicy`? For example: what if `err` is a failure to create the role? We should return a clear error here if the role can not be created?

Reply:
Hey Rob, good catch, I miswrote this statement. I meant to only attempt to attach the policy if there's no error rather than only if there's an error! I think that should cover what you're suggesting here with clear error returns.